SCClassLibrary/Common/Collections/linux/extLinuxPlatform_findTerminalEmulator.sc

```diff
@@ -0,0 +1,38 @@
++ LinuxPlatform {
+
+  *findTerminalEmulator {|action|
+    var supportedTerminalEmulatorsCmds = IdentityDictionary[
```

Comment:
I have 3 comments here:

- I'd prefer consistent symbol notation - in this case ```'symbol'``` and not ```\symbol```
- rethink the order of commands prioritizing more common options
  - for me that's rather theoretical since I'm not using Linux regularly, but my guess is that we should do something like:
    1. universal terminal command first (*)
    2. gnome/kde/other desktop environments
    3. xterm and other popular emulators
- re: (*): I understand that there's no universal terminal command, that would launch terminal of choice on the user machine, but I read that there is `x-terminal-emulator`. IIUC this would be the closest option to "open default terminal", even though it's only for debian, AFAIU.

Reply:
- Consistent symbol notation: agreed!
- Reorder: right now there's no defined order, am I right? They are dictionary keys, so their order is not reliable. I'll think about this, as there should also be a way for users to set their preferred emulator as first priority

Reply:
> * Reorder: right now there's no defined order, am I right? They are dictionary keys, so their order is not reliable. I'll think about this, as there should also be a way for users to set their preferred emulator as first priority

Ah yes, I've read it wrong.
When I was thinking about the implementation, I was considering having `supportedTerminalEmulatorsCmds` as an array of commands, and then use `string.spli($ ).first` or something like that to get the command name itself when iterating to test their validity. That way the commands can be set in preferred order. What do you think about that?

Reply:
Makes sense! Now, three problems:
- x-terminal-emulator: we can't predict the right arguments' syntax for it, since it's a symlink and we don't know what exact program it points to. I think it's better to avoid it.
- terminals without --hold option (xfce, gnome, mate, and terminator): I don't know what's best to do, as I feel this is an important feature of runInTerminal, and appending `;read` to user commands doesn't feel good to me. Worse, the ones without --hold are probably among the most used on debian-based distros. I gave them lower priority for now.
- some terminals seem to need extra quotes around the shell command ( xfce, mate, termite, terminator )... which apparently sends us to an escaping hell... there must be some other way, but I haven't found it yet, just leaving this as a note

Reply:
> * x-terminal-emulator: we can't predict the right arguments' syntax for it, since it's a symlink and we don't know what exact program it points to. I think it's better to avoid it.

Got it... I didn't realize this is just a symlink. Then yes, better to avoid it.

> * terminals without --hold option (...)

What if we went the "temporary script" route, like on macOS? I _think_ this would help with escaping.

> * some terminals seem to need extra quotes around the shell command  (...)

Based on [this](https://stackoverflow.com/questions/3512055/avoid-gnome-terminal-close-after-script-execution) we could consider dropping to `bash` after the command... what do you think?

Reply:
I'm giving it a try with temp script and drop to shell.
- temp script: I think it's nice to help with escaping, and it makes terminal commands easier. Plus it's the same thing we do on macos.
- drop to shell vs. `read`:  dropping to a shell feels a bit too heavy handed. I'm giving it a try, but I think I would prefer to append something like `read -p "Press ENTER to close"`
- I would then remove all --hold flags from other terminal commands, to unify the behavior